abci/abci.go

```diff
@@ -74,9 +74,13 @@ func (h *ProposalHandler) PrepareProposalHandler() sdk.PrepareProposalHandler {
 			return &abci.ResponsePrepareProposal{Txs: make([][]byte, 0)}, err
 		}
 
+		// Get the max gas limit and max block size for the proposal.
+		_, maxGasLimit := proposals.GetBlockLimits(ctx)
+		proposal := proposals.NewProposal(h.logger, req.MaxTxBytes, maxGasLimit)
```

Comment:
Shld we take the min of the req.MaxTxBytes, and the ctx.ConsensusParams().MaxBytes? Or do we know that req.MaxTxBytes is always less?

Reply:
comet deducts the val set size and evidence size before requesting